### PR TITLE
Minor qol/aesthetic changes + link tests

### PR DESCRIPTION
**Pull request by biox**
- Hardcode some permissions so that tests pass with variable umasks
- Rename all custom resources to be consistent
- Add integration test for symlink, corrected a perm test
- Removed comments from .kitchen.yml

### CODE REVIEW

File: resources/vault_install_binary.rb [hunk 1]
```diff
@@ -23,11 +23,13 @@
 default_action :install
 
 action :install do
-  directory new_resource.extract_to do
-    mode '0755'
-    owner new_resource.user
-    group new_resource.group
-    recursive true
```
Comment (jessebolson) [inline resources/vault_install_binary.rb]: Doesn't seem like the cleanest way to do this, my syntaxy checker yelled at me for having >80 characters on that line.

File: resources/vault_install_binary.rb [hunk 2]
```diff
@@ -23,11 +23,13 @@
...
+  [new_resource.extract_to, ::File.join(new_resource.extract_to, new_resource.version)].each do |dir|
+    directory dir do
+      mode '0755'
+      owner new_resource.user
+      group new_resource.group
+      recursive true
+    end
```
Comment (biox) [inline resources/vault_install_binary.rb:36]: Doesn't seem like the cleanest way to do this, my syntaxy checker yelled at me for having >80 characters on line 26.

File: resources/vault_install_binary.rb [hunk 3]
```diff
@@ -37,6 +39,10 @@
     group new_resource.group
   end
 
+  file ::File.join(new_resource.extract_to, new_resource.version, 'vault') do
+    mode '0755'
+  end
```
Comment (biox) [inline resources/vault_install_binary.rb:48]: This also doesn't 'feel' right.

File: test/integration/default/inspec/default_spec.rb [hunk 4]
```diff
@@ -26,14 +26,20 @@
...
 describe service('vault') do
   it { should be_installed }
   it { should be_enabled }
   it { should be_running }
 end
+
+describe file('/usr/local/bin/vault') do
+  it { should exist }
+  it { should be_symlink }
+  it { should be_linked_to '/opt/vault/0.7.2/vault' }
```
Comment (biox) [inline test/integration/default/inspec/default_spec.rb:44]: All of the `0.7.2` values in this file should be variable-ized, but I'm not sure how to do that with custom resources.

File: resources/vault_install_binary.rb (hunk 2 above)
Comment (axelrtgs) [reply resources/vault_install_binary.rb:36]: I don't think you need to create the second sir as the poise extract does it already and sets perms.

Comment (biox) [reply resources/vault_install_binary.rb:36]: While the poise extract does create the files and set the owner/group, it does not catch the mode which made one of our tests fail. How do you think that should be handled?

Comment (axelrtgs) [reply resources/vault_install_binary.rb:36]: Good question. Let's try and find a solution with the group. I'm collecting luggage and heading home to sleep. Will find some time tomorrow to check back in after I get up and figure out home.

**Review by axelrtgs — APPROVED** [review]

**Pull request merged**